merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/GetSimulationResultsAction.java

```diff
@@ -97,9 +98,10 @@ public Map<String, List<Violation>> getViolations(final PlanId planId, final Sim
       throw new RuntimeException("Assumption falsified -- mission model for existing plan does not exist");
     }
 
-
     final var activities = new ArrayList<ActivityInstance>();
-    for (final var entry : results.simulatedActivities.entrySet()) {
+    final var simulatedActivities = this.simulationService.get(planId, revisionData,
+        r -> r.simulatedActivities).orElseGet(Map::of);
```

Comment:
This formatting splits a logical method call and simultaneously groups half of it with a separate method call. Maybe something like this makes the logic flow more clearly?

```suggestion
    final var simulatedActivities = this.simulationService
        .get(planId, revisionData, r -> r.simulatedActivities)
        .orElseGet(Map::of);
```

Reply:
Good call, latter is much clearer

Reply:
Change has been made.